app/code/Bulbulatory/Recomendations/view/adminhtml/ui_component/bulbulatory_recomendations_recommendations_listing.xml

```diff
@@ -0,0 +1,142 @@
+<listing xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+    xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Ui:etc/ui_configuration.xsd">
+    <argument name="data" xsi:type="array">
+        <item name="js_config" xsi:type="array">
+            <item name="provider" xsi:type="string">
+                bulbulatory_recomendations_recommendations_listing.bulbulatory_recomendations_recommendations_listing_data_source
+            </item>
+            <item name="deps" xsi:type="string">
+                bulbulatory_recomendations_recommendations_listing.bulbulatory_recomendations_recommendations_listing_data_source
+            </item>          
+        </item>
+        <item name="spinner" xsi:type="string">
+            spinner_columns
+        </item>
+    </argument>
+    <dataSource name="nameOfDataSource">
+        <argument name="dataProvider" xsi:type="configurableObject">
+            <argument name="class" xsi:type="string">Magento\Framework\View\Element\UiComponent\DataProvider\DataProvider</argument>
+            <argument name="name" xsi:type="string">bulbulatory_recomendations_recommendations_listing_data_source</argument>
+            <argument name="primaryFieldName" xsi:type="string">recommendation_id</argument>
+            <argument name="requestFieldName" xsi:type="string">recommendation_id</argument>
+            <argument name="data" xsi:type="array">
+                <item name="config" xsi:type="array">
+                    <item name="component" xsi:type="string">Magento_Ui/js/grid/provider</item>
+                    <item name="update_url" xsi:type="url" path="mui/index/render" />
+                    <item name="storageConfig" xsi:type="array">
+                        <item name="indexField" xsi:type="string">id</item>
```

Comment:
Tutaj powinna być nazwa kolumny primary, w Twoim przypadku jest to `recommendation_id`. Teraz nie ma to znaczenia, ponieważ nie jest to wykorzystywane, ale w jednym z kolejnych zadań będzie miało i niepoprawna wartość może spowodować sporą stratę czasu na szukanie, dlaczego coś nie działa.